packages/salesforcedx-vscode-core/src/commands/templates/baseTemplateCommand.ts

```diff
@@ -0,0 +1,160 @@
+      execution.command.toString(),
+      (execution.stderrSubject as any) as Observable<Error | undefined>
+    );
+    channelService.streamCommandOutput(execution);
+    ProgressNotification.show(execution, cancellationTokenSource);
+    taskViewService.addCommandExecution(execution, cancellationTokenSource);
+  }
+
+  private identifyDirType(outputDirectory: string): string {
+    const defaultDirectoryPath = path.join(
+      SelectOutputDir.defaultOutput,
+      this.getDefaultDirectory()
+    );
+    return outputDirectory.endsWith(defaultDirectoryPath)
+      ? 'defaultDir'
+      : 'customDir';
+  }
+
+  private getPathToSource(outputDir: string, fileName: string): string {
+    const sourceDirectory = path.join(getRootWorkspacePath(), outputDir);
+    return this.sourcePathStrategy.getPathToSource(
+      sourceDirectory,
+      fileName,
+      this.getFileExtension()
+    );
+  }
+
+  protected abstract sourcePathStrategy: SourcePathStrategy;
+
+  protected abstract getFileExtension(): string;
+
+  protected abstract getDefaultDirectory(): string;
+}
+
+export interface SourcePathStrategy {
+  getPathToSource(dirPath: string, fileName: string, fileExt: string): string;
+}
+
+export class DefaultPathStrategy implements SourcePathStrategy {
+  public getPathToSource(
+    dirPath: string,
+    fileName: string,
+    fileExt: string
+  ): string {
+    return path.join(dirPath, `${fileName}${fileExt}`);
+  }
+}
+export class BundlePathStrategy implements SourcePathStrategy {
+  public getPathToSource(
+    dirPath: string,
+    fileName: string,
+    fileExt: string
+  ): string {
+    const bundleName = fileName;
+    return path.join(dirPath, bundleName, `${fileName}${fileExt}`);
+  }
+}
+
+export class FilePathExistsChecker
+  implements PostconditionChecker<DirFileNameSelection> {
+  private fileExtensionsToCheck: string[];
+  private sourcePathStrategy: SourcePathStrategy;
+  private metadataLabel: string;
+  public constructor(
+    fileExtensionsToCheck: string[],
+    sourcePathStrategy: SourcePathStrategy,
+    metadataLabel: string
+  ) {
+    this.fileExtensionsToCheck = fileExtensionsToCheck;
+    this.sourcePathStrategy = sourcePathStrategy;
+    this.metadataLabel = metadataLabel;
+  }
+
+  public async check(
+    inputs: ContinueResponse<DirFileNameSelection> | CancelResponse
+  ): Promise<ContinueResponse<DirFileNameSelection> | CancelResponse> {
+    if (inputs.type === 'CONTINUE') {
+      const outputDir = inputs.data.outputdir;
+      const fileName = inputs.data.fileName;
+      const files = await vscode.workspace.findFiles(
+        this.createFilesGlob(outputDir, fileName)
+      );
+      // If file does not exist then create it, otherwise prompt user to overwrite the file
+      if (files.length === 0) {
+        return inputs;
+      } else {
+        const overwrite = await notificationService.showWarningMessage(
+          nls.localize('warning_prompt_file_overwrite', this.metadataLabel),
```

Comment:
<img width="316" alt="Screen Shot 2019-04-09 at 4 52 39 PM" src="https://user-images.githubusercontent.com/1639375/55842343-ea846080-5ae7-11e9-8eaf-60732f461d14.png">
The new warning message includes the metadata type of the entity that you are choosing to overwrite. The new string added to the end of the i18n.ts file with names ending in `*_message_name` represent the different metadata names that could replace 'Aura Bundle' in the screenshot.